evi-next-js-function-calling/src/components/ClientComponent.tsx

```diff
@@ -0,0 +1,95 @@
+"use client";
+import { VoiceProvider, ToolCall, ToolResponse, ToolError } from "@humeai/voice-react";
+import Messages from "./Controls";
+import Controls from "./Messages";
+
+const handleToolCall = async (
+  toolCall: ToolCall
+): Promise<ToolResponse | ToolError> => {
+  console.log("Tool call received", toolCall);
+
+  if (toolCall.name === 'weather_tool') {
+    try {
+      const args = JSON.parse(toolCall.parameters) as {
+        location: string;
+        format: 'fahrenheit' | 'celsius';
+      };
+
+      const location = await fetch(
+        `https://geocode.maps.co/search?q=${args.location}&api_key=${process.env.NEXT_PUBLIC_GEOCODING_API_KEY}`,
+      );
+
+      const locationResults = (await location.json()) as {
+        lat: string;
+        lon: string;
+      }[];
+
+      const { lat, lon } = locationResults[0];
+
+      const pointMetadataEndpoint: string = `https://api.weather.gov/points/${parseFloat(lat).toFixed(3)},${parseFloat(lon).toFixed(3)}`;
+
+      const result = await fetch(pointMetadataEndpoint, {
+        method: 'GET',
+      });
+
+      const json = (await result.json()) as {
+        properties: {
+          forecast: string;
+        };
+      };
+      const { properties } = json;
+      const { forecast: forecastUrl } = properties;
+
+      const forecastResult = await fetch(forecastUrl);
+
+      const forecastJson = (await forecastResult.json()) as {
+        properties: {
+          periods: unknown[];
+        };
+      };
+      const forecast = forecastJson.properties.periods;
+
+      return {
```

Comment:
using the `response.success()` and `response.error()` helpers here would be useful for documentation https://github.com/HumeAI/empathic-voice-api-js/pull/191